tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerTransaction.java

```diff
@@ -203,7 +203,7 @@ protected void doCommit() throws TransactionException {
 
             final Set<TinkerElementContainer> readElements = txReadElements.get();
             if (readElements != null)
-                readElements.stream().forEach(e -> e.reset());
+                readElements.stream().forEach(e -> e.commit(txVersion));
```

Comment:
Thanks Cole for figuring this out! 

I think it would be good to add a test to verify case when element is in 
both `changedVertices` and `readElements`, that `usedInTransactions` counter will be reduced exactly 2 times

Reply:
Thanks for the suggestion, I've added such a test